src/database.py

```diff
@@ -136,18 +149,64 @@ def set_permissions(db, user):
         template = "Unhandled permission type '{permissions}' for user '{name}'"
         raise Exception(template.format(name=user.name, permissions=user.permissions))
 
-
-def migrate_schema(db_password):
+def migrate_schema(user, password, host=None, config_file=None,
+                   placeholders=None):
     image = get_image_name("montagu-migrate", versions.db)
     pull(image)
-    run([
-        "docker", "run",
-        "--rm",
-        "--network=" + network_name,
-        image,
-        "migrate", "-user=" + root_user, "-password=" + db_password
-    ], check=True)
-
+    cmd = ["docker", "run", "--rm", "--network=" + network_name, image]
+    if host:
+        cmd += ["-url=jdbc:postgresql://{}/montagu".format(host)]
+    if config_file:
+        cmd += ["-configFile=" + config_file]
+    if placeholders:
```

Comment:
my feeling is that `migrate_schema_core` and `migrate_schema_annex` are sufficiently different that there's no point in them both calling through to the same function like this - it's not very easy to see what each is doing